app/page/send/index.tsx

```diff
@@ -149,26 +152,26 @@ const SendAmount = ({
   const { token, txTo } = txInfo
 
   const { provider } = useContext(ProviderContext)
-  const network = useNetwork()
 
   const [inputAmount, setInputAmount] = useState<string>("0")
+  const [transferAmount, setTransferAmount] = useState(0n)
 
   const handleAmountChange = async (event: ChangeEvent<HTMLInputElement>) => {
     const value = event.target.value
     setInputAmount(value)
+    setTransferAmount(ethers.parseUnits(value, token.decimals))
   }
 
   const handleSend = useCallback(async () => {
-    const signer = await provider.getSigner()
     if (
       address.isEqual(
         token.address,
         "0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE"
       )
     ) {
-      return sendNativeToken(signer, txTo, inputAmount)
+      return sendNativeToken(provider, txTo, transferAmount)
     }
-    return sendErc20Token(signer, txTo, inputAmount, token)
+    return sendErc20Token(provider, txTo, transferAmount, token)
   }, [txTo, inputAmount])
```

Comment:
Not sure if it's better to add the `transferAmount` dependency?

Reply:
Though they should always be changed together, it's fine to add it for safety.